IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.h

```diff
@@ -163,15 +163,30 @@
 - (BOOL)saveWipeInfoWithContext:(nullable id<MSIDRequestContext>)context
                           error:(NSError * _Nullable * _Nullable)error;
 
+/*
+ Saves app metadata
+ */
 - (BOOL)saveAppMetadata:(nonnull MSIDAppMetadataCacheItem *)metadata
                 context:(nullable id<MSIDRequestContext>)context
                   error:(NSError * _Nullable * _Nullable)error;
 
 /*
- Gets app metadata for a particular key
+ Gets app metadata for a particular query
+ */
+- (nullable MSIDAppMetadataCacheItem *)getAppMetadataWithQuery:(nonnull MSIDAppMetadataCacheQuery *)query
+                                                       context:(nullable id<MSIDRequestContext>)context
+                                                         error:(NSError * _Nullable * _Nullable)error;
+
+/*
+ Remove app metadata
  */
-- (nullable MSIDAppMetadataCacheItem *)getAppMetadata:(nonnull MSIDAppMetadataCacheKey *)key
-                                              context:(nullable id<MSIDRequestContext>)context
-                                                error:(NSError * _Nullable * _Nullable)error;
+- (BOOL)removeAppMetadata:(nonnull MSIDAppMetadataCacheItem *)appMetadata
+                  context:(nullable id<MSIDRequestContext>)context
+                    error:(NSError * _Nullable * _Nullable)error;
+
+
+- (nullable NSArray<MSIDAppMetadataCacheItem *> *)getAppMetadataEntries:(nonnull MSIDAppMetadataCacheQuery *)query
```

Comment:
Please combine this with getAppMetadataWithQuery. 
For a single entry result, just return with a single item maybe?

Reply:
Done. I updated the code to use similar logic by getting all metadata entries and then returning the one with the match, but still kept both api's for returning a single item and returning all items as I need it for test app.